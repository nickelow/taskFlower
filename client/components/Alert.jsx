import React from 'react';
const styles = {width: '400px', backgroundColor: 'YellowGreen'}


class Alert extends React.Component {
	constructor(props){
		super(props)
	}
	render(){
		return(
			<div class="ui message" stlye={styles}>
				<i class="close icon" onClick={this.props.dismissAlert}></i>
				<p>The list was updated and saved.</p>
			</div>
		)
	}
}

export default Alert