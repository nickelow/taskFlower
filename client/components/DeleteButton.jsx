import React from 'react';


const styles = {
	'position': 'relative',
	'top': '20px',
	'left': '98%'
}
class DeleteButton extends React.Component {
	constructor(props) {
	  super(props);
	}
	render() {
	  return(
      <i class="fa fa-trash" aria-hidden="true" onClick= {this.props.deleteCard}  id= {this.props.id} style={styles}></i>
	  )
	}
}

export default DeleteButton