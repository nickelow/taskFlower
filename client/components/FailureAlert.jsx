import React from 'react';

class FailureAlert extends React.Component {
	constructor(props){
		super(props)
	}
	render(){
		return(
			<div class="ui message">
				<i class="close icon" onClick={this.props.dismissAlert}></i>
				<p>Sorry, something went wrong. Please try again.</p>
			</div>
		)
	}
}

export default FailureAlert