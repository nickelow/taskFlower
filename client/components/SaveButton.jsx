import React from 'react';
import reactDOM from 'react-dom';

class SaveButton extends React.Component {
  constructor(props){
    super(props)
    
  }

  render(){
    return(
        <button class="ui active button" onClick={this.props.saveCards}>
          <i class="user icon"></i>
          Save
        </button>
    ) 
  }

}

export default SaveButton;