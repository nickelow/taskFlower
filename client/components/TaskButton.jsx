import React from 'react';

class TaskButton extends React.Component {
  constructor(props){
    super(props)
    
  }

  render(){
    return(
        <button class="ui right labeled icon button" onClick={this.props.addCard}>
          <i class="fa fa-plus" aria-hidden="true"></i>
             New Task
        </button>
    ) 
  }

}

export default TaskButton;