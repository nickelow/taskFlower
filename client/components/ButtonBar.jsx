import React from 'react';
import TaskButton from './TaskButton.jsx';
import SaveButton from './SaveButton.jsx';
import DisabledSaveButton from './DisabledSaveButton.jsx'

import Alert from "./Alert.jsx";

const styles = {'position': 'relative', 'top': '20%', 'left': '60%', 'width': '400px', 'height': '100px', 'display': 'inline-block'}
const headerStyles ={'position': 'absolute', 'left': '5%', 'top': '8%',  'width': '40%'}
const barStyles = {'diplay': 'inline-block', 'verticalAlign': 'middle'};

class ButtonBar extends React.Component {
	constructor(props) {
		super(props)
		this.state={
			modified:this.props.modified
		}
	}
	 componentWillReceiveProps(){
    this.setState({
      modified: this.props.modified
    })
  }
	render(){
		return([<div style={barStyles}>     <div class="ui header" style={headerStyles}>
        <i class="fa fa-list-alt fa-4x"></i>
        <div class="content" style={{'padding': '0.5rem 1rem', 'top': '50%'}}>
          Tasks
          <div class="sub header">Add and edit your tasks</div>
        </div>
      </div>,
					    <div class="ui buttons" style={styles}>
					      <div className="buttons"><TaskButton addCard={this.props.addCard}/></div>
					      <div className="buttons">{this.props.modified === false ? <DisabledSaveButton/> : <SaveButton saveCards={this.props.saveCards}/>}</div>
						  </div></div>]
		
		)
	}
}

export default ButtonBar;

