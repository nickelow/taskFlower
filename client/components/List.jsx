import React from 'react';
import reactDOM from 'react-dom';
import Card from './Task.jsx';
import { DragDropContext } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'
import update from 'immutability-helper';
import TaskButton from './TaskButton.jsx';
import DeleteButton from './DeleteButton.jsx';
import ButtonBar from './ButtonBar.jsx';
import Alert from './Alert.jsx';
import axios from 'axios';

const serverPath = 'http://cfassignment.herokuapp.com/matthewnicolaou/tasks'

const cardStyle = {
	'width': '80%'
}

const listStyle = {
	'marginRight': '5%',
	'marginLeft': '5%',
	'marginTop': '5%',
	backgroundColor: '#f2f9fd',
	'position': 'static'
}



class List extends React.Component {
  constructor(props) {
    super(props);
    this.moveCard = this.moveCard.bind(this);
    this.findCard = this.findCard.bind(this);
    this.editMode = this.editMode.bind(this);
    this.editCard = this.editCard.bind(this);
    this.addCard = this.addCard.bind(this);
    this.saveCards = this.saveCards.bind(this);
    this.getCards = this.getCards.bind(this);
    this.deleteCard = this.deleteCard.bind(this);
    this.dismissAlert = this.dismissAlert.bind(this);
    this.state = {
      cards: [],
      modified: false,
      saved: false,
      target: null,
      ids: 0
    };
  }

  componentDidMount(){
  	this.getCards()
  }

  findCard(id) {
    const { cards } = this.state;
    const card = cards.filter(c => c.id === id)[0];

    return {
      card,
      index: cards.indexOf(card)
    };
  }

	moveCard(dragIndex, hoverIndex) {
		const { cards } = this.state
		const dragCard = cards[dragIndex]

		this.setState(
			update(this.state, {
				cards: {
					$splice: [[dragIndex, 1], [hoverIndex, 0, dragCard]],
				},
			}),
		)
		this.setState({
      modified: true
    })
	}

	saveCards(){
		var context = this;
		var taskList = this.state.cards;
		axios.post(serverPath, {tasks: taskList})
		.then(function(response){
			console.log('sent to server')
			context.setState({
				modified: false,
				saved: true
			})
		})
		.catch(function(error){
			console.log('error', error)
      this.setState({
        saved: false
      })
		})
	}

  getCards(){
  	var data = [];
  	var context = this;
    axios.get(serverPath)
    .then(function(response){
      console.log(response.data)
      data = response.data.tasks
      context.setState({
    	  cards: data
      })
    })
    .catch(function(err){
      console.log(err)
      this.setState({
        saved: false
      })
    })
  }

	addCard(){
		this.setState(
			this.state.cards.splice(0, 0, 
				{id: this.state.cards.length, 
				 text: 'New Task'})
			)
		this.setState({
					modified: true
				})
	}

  deleteCard(index){
  	var cardId = index.target.id
  	var newList = this.state.cards.filter(function(card){
      	return card.text !== cardId
      })
  	this.setState({
      cards: newList,
      modified: true
  	})
  }


	editMode(event){
		console.log('Matt is a baller', event.target)
		this.setState({
			modified: true
		})
	}

	editCard(e){
		console.log(e.target.value)
		var cardId = e.target.id
		var cardList = this.state.cards
		console.log(cardId);
		cardList[cardId] = {text:e.target.value, id: cardId}
		console.log(cardList)
    this.state.cards[cardId].text = e.target.value
    this.setState({
    	modified: true
    })
	}

	dismissAlert(){
		this.setState({
			saved: false
		})
	}

  render() {
    const { cards } = this.state;
    return ([
    	<div style={listStyle}>
    	<div><ButtonBar addCard={this.addCard} modified={this.state.modified} saveCards={this.saveCards}/></div>
      <div style={cardStyle}>
      {this.state.saved === true ? <Alert dismissAlert={this.dismissAlert}/> : <div></div>}
        {cards.map((card, i) => {
          return (
            <Card key={i}
	            index={i}
	            id={i}
	            text={card.text}
	            moveCard={this.moveCard} 
	            findCard={this.findCard}
	            editMode={this.editMode}
	            deleteCard={this.deleteCard}
	            editCard ={this.editCard}
	           />     
          );
        })}
      </div>
      </div>
    ]);
  }
}

export default DragDropContext(HTML5Backend)(List)

