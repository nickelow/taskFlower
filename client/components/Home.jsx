import React from 'react';
import reactDOM from 'react-dom';
import { Button } from 'react-bootstrap';
import axios from 'axios';
import List from './List.jsx'

const wellStyles = {maxWidth: 400, margin: '0 auto 10px'};
const listStyles = {"font-family": "Helvetica, Arial, sans-serif", "font-size": "13px", "color": "rgb(51, 51, 51)"}
const headerStyles ={'position': 'relative', 'left': '10%', 'top': '20%', 'width': '40%'}
const serverPath = 'http://cfassignment.herokuapp.com/matthewnicolaou/tasks'


class Home extends React.Component {
	constructor(props){
    super(props);
    this.state = {
      tasks: []
    }
    this.getTasks = this.getTasks.bind(this);
    this.sendResume = this.sendResume.bind(this);
  }

  getTasks(){
    axios.get(serverPath)
    .then(function(response){
      console.log(response.data)
    })
    .catch(function(err){
      console.log(err)
    })
  }

  sendResume(){
    axios.post(closePath, closeApp)
    .then(function(response){
      console.log(response)
    })
    .catch(function(err){
      console.log(err)
    })
  }

  render(){
    return([
     <span><List style={listStyles}/></span>])
  }
}

export default Home; 

