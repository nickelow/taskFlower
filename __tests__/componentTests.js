import React from 'react'
import { shallow, mount } from 'enzyme';
import { DragDropContext } from 'react-dnd'
import HTML5Backend from 'react-dnd-html5-backend'
import Home from '../client/components/Home.jsx'
import List from '../client/components/List.jsx'
import Card from '../client/components/Task.jsx'
import TaskButton from '../client/components/TaskButton.jsx';
import DeleteButton from '../client/components/DeleteButton.jsx';
import SaveButton from '../client/components/SaveButton.jsx';
import ButtonBar from '../client/components/ButtonBar.jsx';
import Alert from '../client/components/Alert.jsx';
import FailureAlert from '../client/components/FailureAlert.jsx';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

describe('<Home />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<List />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<List />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


describe('<TaskButton/>', () => {
  let wrapper;

  beforeEach(() => {
  	const spy = sinon.spy();
  	const list = shallow(<List/>)
    wrapper = shallow(<TaskButton addCard={spy}/>);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<DeleteButton/>', () => {
  let wrapper;
  let  component;
  beforeEach(() => {
  	const spy = sinon.spy();
    wrapper = shallow(<DeleteButton deleteCard={spy}/>);
    component = wrapper.instance()
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

describe('<SaveButton/>', () => {
  let wrapper;

  beforeEach(() => {
  	const spy = sinon.spy();
    wrapper = shallow(<SaveButton saveCards={spy}/>);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

describe('<ButtonBar/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ButtonBar />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('<Alert />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Alert />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});