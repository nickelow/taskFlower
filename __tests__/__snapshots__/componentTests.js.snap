// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render 1`] = `
<div
  className="ui message"
  stlye={
    Object {
      "backgroundColor": "YellowGreen",
      "width": "400px",
    }
  }
>
  <i
    className="close icon"
  />
  <p>
    The list was updated and saved.
  </p>
</div>
`;

exports[`<ButtonBar/> should render 1`] = `
Array [
  <div
    style={
      Object {
        "diplay": "inline-block",
        "verticalAlign": "middle",
      }
    }
  />,
]
`;

exports[`<DeleteButton/> should render 1`] = `
<i
  aria-hidden="true"
  className="fa fa-trash"
  onClick={[Function]}
  style={
    Object {
      "left": "98%",
      "position": "relative",
      "top": "20px",
    }
  }
/>
`;

exports[`<Home /> should render 1`] = `
Array [
  <span />,
]
`;

exports[`<List /> should render 1`] = `<List />`;

exports[`<SaveButton/> should render 1`] = `
<button
  className="ui active button"
  onClick={[Function]}
>
  <i
    className="checkmark icon"
  />
  Save
</button>
`;

exports[`<TaskButton/> should render 1`] = `
<button
  className="ui right labeled icon button"
  onClick={[Function]}
>
  <i
    aria-hidden="true"
    className="plus icon"
    onClick={[Function]}
  />
  New Task
</button>
`;
